src/Data/Expenses/Ledger/Xml.hs

```diff
@@ -0,0 +1,202 @@
+-- | Data type for the @\<amount\>@ element in a ledger XML document.
+data Amount = Amount { amountCurrency :: String, amountQuantity :: D.Decimal }
+  deriving (Show, Eq)
+
+
+
+-- | Data type for the @\<posting\>@ element in a ledger XML document.
+data Posting = Posting { postingAccount :: String, postingAmount :: Amount }
+  deriving (Show, Eq)
+
+
+
+-- | Data type for the @\<transaction\>@ element in a ledger XML document.
+data Transaction = Transaction
+  { transactionDate :: String
+  , transactionPayee :: String
+  , transactionPostings :: [Posting]
+  }
+  deriving (Show, Eq)
+
+
+
+getChildText :: HXT.ArrowXml a => String -> a XmlTree String
+getChildText child =
+  proc f -> do
+    node <- HXT.hasName child <<< HXT.getChildren -< f
+    s <- HXT.getText <<< HXT.getChildren -< node
+    HXT.returnA -< s
+
+
+
+getNestedText :: HXT.ArrowXml a => String -> String -> a XmlTree String
+getNestedText child1 child2 =
+  proc f -> do
+    node1 <- HXT.hasName child1 <<< HXT.getChildren -< f
+    node2 <- HXT.hasName child2 <<< HXT.getChildren -< node1
+    s <- HXT.getText <<< HXT.getChildren -< node2
+    HXT.returnA -< s
+
+
+
+-- | Results in a parsed number, or empty if the string is not numeric.
+textAsDecimal :: HXT.ArrowList a => a String D.Decimal
+textAsDecimal =
+  HXT.arrL $ maybeToList . readMaybe
+
+
+
+-- | An 'Amount' from an @\<amount\>@ XML element.
+amount  :: HXT.ArrowXml a => a XmlTree Amount
+amount =
+  HXT.hasName "amount" >>>
+  proc a -> do
+    amtCurrency <- getNestedText "commodity" "symbol" -< a
+    amtQuantity <- textAsDecimal <<< getChildText "quantity" -< a
+    HXT.returnA -< Amount { amountCurrency = amtCurrency
+                          , amountQuantity = amtQuantity
+                          }
+
+
+
+-- | An 'Amount' from a @\<posting\>@ XML element.
+postAmount :: HXT.ArrowXml a => a XmlTree Amount
+postAmount =
+  HXT.hasName "posting" />
+  HXT.hasName "post-amount" />
+  HXT.hasName "amount" >>>
+  amount
+
+
+
+-- | 'Posting's from a given @\<transaction\>@ element.
+postings :: HXT.ArrowXml a => a XmlTree [Posting]
+postings =
+  HXT.listA $
+  HXT.hasName "transaction" />
+  HXT.hasName "postings" />
+  HXT.hasName "posting" >>>
+  proc p -> do
+    pstAccount <- getNestedText "account" "name" -< p
+    pstAmount <- postAmount -< p
+    HXT.returnA -< Posting { postingAccount = pstAccount
+                           , postingAmount = pstAmount
+                           }
+
+
+
+-- | 'Transaction' from a @\<transaction\>@ element.
+transaction :: HXT.ArrowXml a => a XmlTree Transaction
+transaction =
+  HXT.hasName "transaction" >>>
+  proc txn -> do
+    txnPayee <- getChildText "payee" -< txn
+    txnDate <- getChildText "date" -< txn
+    txnPostings <- postings -< txn
+    HXT.returnA -< Transaction { transactionDate = txnDate
+                               , transactionPayee = txnPayee
+                               , transactionPostings = txnPostings
+                               }
```

Comment:
Style inconsistency. The type constructor and the `{}` braces are aligned in the code below.